### PR TITLE
Implement URL search-replace for linkchecker

### PR DESCRIPTION
**Pull request by membralala**
### Short description

This PR provides a basic search and replace solution for URLs. It therefore enhances the linkchecker's list view

### Proposed changes

- Add an URL edit form to the linkchecker's list view
- Replace the URL in pages contents 

### Resolved issues

Fixes: #1024

### CODE REVIEW

**Comment by membralala** [comment]
With the way this is working now, I noticed an issue, that the replacement fails due to the form validation, if the URL, that you want to use is already in the database and I'm not quite sure, how to tackle this one. Could we even ignore the validation as there is already some browser-side validation and the actual model is never updated anyway?

File: integreat_cms/locale/de/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -4301,10 +4302,14 @@ msgstr "Anzahl an Regionen"
...
 msgstr "Noch keine Sprachen vorhanden."
 
-#: cms/templates/linkcheck/link_list_row.html:24
+#: cms/templates/linkcheck/link_list_row.html:25
 msgid "Go to source"
 msgstr "Inhalt aufrufen"
 
+#: cms/templates/linkcheck/link_list_row.html:35
+msgid "Edit URL"
+msgstr "URL bearbeiten"
```
Comment (timoludwig) [inline integreat_cms/locale/de/LC_MESSAGES/django.po:4311]:
```suggestion
msgid "Replace URL globally"
msgstr "URL zentral ersetzen"
```

File: integreat_cms/cms/views/linkcheck/linkcheck.py [hunk 2]
```diff
@@ -6,9 +8,10 @@
 from django.views.generic import ListView
 from django.views.generic.base import RedirectView
 
-from linkcheck.models import Link
+from linkcheck.models import Link, Url
 
 from ...utils.filter_links import filter_links
+from ...forms.linkchecker.edit_url_form import EditUrlForm
```
Comment (timoludwig) [inline integreat_cms/cms/views/linkcheck/linkcheck.py:14]:
```suggestion
from ...forms.linkcheck.edit_url_form import EditUrlForm
```
I'd rename the directory from `linkchecker` to `linkcheck` to be more consistent with the rest of our linkcheck files.

File: integreat_cms/cms/views/linkcheck/linkcheck.py [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import re
+
 from django.conf import settings
 from django.contrib import messages
 from django.urls import reverse
```
Comment (timoludwig) [inline integreat_cms/cms/views/linkcheck/linkcheck.py:5]:
```suggestion
from django.urls import reverse
from django.shortcuts import get_object_or_404, redirect
```

**Review by timoludwig — COMMENTED** [review]
Thanks a lot! :muscle: 
In general, it works as expected, but I noticed a few problems: It replaces the content of all versions, which means a lot of overhead for replaced links that are never actually used or even viewed in the link list. It turns out my initial idea didn't work as I thought, so I came up with a different idea to filter only the latest versions of the translations, see my comments below for details.

> With the way this is working now, I noticed an issue, that the replacement fails due to the form validation, if the URL, that you want to use is already in the database and I'm not quite sure, how to tackle this one. Could we even ignore the validation as there is already some browser-side validation and the actual model is never updated anyway?

I suggest to circumvent this problem by using a basic `forms.Form` instead of `forms.ModelForm`, but see my comments below for details.

File: integreat_cms/cms/urls/protected.py [hunk 4]
```diff
@@ -590,8 +590,32 @@
...
+                                                                    linkcheck.LinkListView.as_view(),
+                                                                    name="edit_link",
+                                                                ),
+                                                                path(
+                                                                    "edit/",
+                                                                    linkcheck.LinkListView.as_view(),
+                                                                    name="edit_url",
+                                                                ),
+                                                            ]
+                                                        ),
```
[the remark this answers is not in the document]
Comment (membralala) [reply integreat_cms/cms/urls/protected.py:615]: Indeed!

**Comment by membralala** [comment]
Thanks for your feedback! I applied your suggestions and rebased the branch.

**Comment by membralala** [comment]
Thanks for your feedback! I applied your suggestions and rebased the branch.

**Comment by timoludwig** [comment]
Oh, and just one additional request: could you add a note to the changelog file? Thanks!

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 UNRELEASED
 ----------
-
+* [ [#1024](https://github.com/digitalfabrik/integreat-cms/issues/) ] Add URL search-replace for linkchecker
```
Comment (timoludwig) [inline CHANGELOG.md:3]: Could you add this to the last line? :see_no_evil:

**Pull request merged**